.vscode/settings.json

```diff
@@ -0,0 +1,7 @@
+{
```

Comment:
Won't checking this file into the repo cause everybody to override it and commit (by mistake for example) their personal preferences?

Reply:
I think personal settings are set in the global `settings.json` and then VSCode takes care of mergin those with the workspace settings, so unless someone actively makes changes to this file, it shouldn't change.

Reply:
But people (me for example) may have workspace-specific settings they want to use.

Reply:
Francesco is correct, the workspace settings checked in here will only override your global settings.
So you personal preferences can go into the `User` global settings.
If you have some good workspace settings then the rest of the team could benefit also by adding them here. What do you think?

Reply:
> Francesco is correct, the workspace settings checked in here will only override your global settings.

Well, I'm not sure that's a good thing. For example, someone (from the team or from the outside) may not like formatting on save for some reason but they would be forced to either use it or keep enabling and disabling it to not check it into source control. I believe IDE settings are a pretty personal thing and it's easier if everybody just does them for themselves. Enabling format on save for a team member is a 10-second task and we wouldn't need to worry about someone committing their changes, different preferences, etc.

I don't use VSCode myself most of the time so I don't have that strong of an opinion but I believe it's a good practice to keep specific IDEs out of the source.

Reply:
As I was initially hesitant to add this in the first place and your last argument, decided to revert it https://github.com/Joystream/joystream/pull/891/commits/ad88e6550203b8df508199f50f6365fd052d81e1

Reply:
Happy to report the issue with eslint plugin in VSCode looking for tsconfig.json at the root is resolved by adding the following in the workspace settings.json:

```json
  "eslint.workingDirectories": [
    "./cli",
    "./pioneer",
    "./tests/network-tests",
    "./types"
  ]
```
as advised by @Gamaranto 

Is this a legit reason to keep .vscode/settings.json in the repo?

Alternative would be to just have a script that injects these settings into the workspace.